netdata/plan.sh

```diff
@@ -67,5 +77,9 @@ do_install() {
     -print \
     -exec bash -c 'sed -e "s#\#\!/usr/bin/env bash#\#\!$1/bin/bash#" --in-place "$2"' _ "$(pkg_path_for bash)" "{}" \;
 
+  build_line "Installing python dependencies"
+  pip install "mysqlclient"
```

Comment:
I love the ideal of including mysql by default. I currently do this in wrapper plans, but I think the extra bloat is worth it for something so common.

On this line, I wonder if we're doing the dependency tracking backwards? Should we be shipping a requirements.txt rather than generating one? Ideally the build should be reproducible.

Reply:
This is a weird case, since it's not a python project, but includes a python module that users can add additional modules to

The python dependencies are variable based on what modules you're using and what user scripts you plug in, and they expect you to just have any you're going to use in your environment. There is no correct and full list of dependencies, and I don't think we'd want to tax users with every possible dependency bundled upfront

I don't do any python development directly, so this is really just my best guess at a reasonable way to do things in this situation that seems to get the job done. I don't have any strong opinion on the right way to do it

Reply:
Thats fair comment, regarding not having a strong opinion. I'm still on the fence regarding this one, but tend to lean towards not accepting, purely because the `mysqlclient` dep is not pinned in any way, and different build runs could produce different packages as a result.

I'll tag this as needs discussion for the moment, and see what kind of feedback others have to offer.

Reply:
Sorry for taking so long to respond to this.   In general, we have been trying to ship packages in the core origin that are as close to the upstream defaults as is reasonable to make them.   This usually means not shipping plugins.  

I'm not familiar with the use of this piece of software, so I would want to get some clarification on that to help determine the best path forward:

Is this software functional without that plugin?
Does upstream recommend shipping with it/enable it by default?
Can we ship it as an "add-on" package, like we do with [sqitch_pg](https://github.com/habitat-sh/core-plans/tree/master/sqitch_pg)?

If we do determine that the correct course of action is to ship this plugin as part of this package, I agree with @predominant  that we should include the `requirements.txt` as part of our source.

Reply:
Thank for the feedback @smacfarlane 

## Is this software functional without that plugin?

The software is at least semi-functional without this plugin, but a primary aspect of the software's functionality is the ability to connect official, community, and your own plugins. Plugins are necessary to add services to monitor

Most plugins are implemented as dependents of a python meta-plugin that lets you hook in new data collectors by writing or dropping in python scripts. 

This pull request accomplishes three things:

1. Sets up the environment for the python meta-plugin and all builtin and user plugins based on it to run
2. Demonstrates how to add optional/custom dependencies to the Python environment that user python plugins run under
3. Provides a test case for verifying that a python user plugins with python dependencies can getting the right environment

## Does upstream recommend shipping with it/enable it by default?

Upstream ships with all official plugins as examples and they self-enable based on whether they can run and find data. [Upstream provides a list of optional dependencies](https://docs.netdata.cloud/packaging/installer/#prepare-your-system) that enable whatever plugins depend on them when installed. AFAIK though, the only way to get any of these optional dependencies into the environment under Habitat is to incorporate them into `plan.sh` like I have here, so users of this package will need a model for how to add python deps to the environment at build time.

## Can we ship it as an "add-on" package, like we do with sqitch_pg?

I don't know that an "add-on" package would make sense, because this PR isn't really adding a mysql plugin--that ships with the upstream package as one of the stock/example python plugins-- but rather adding the optional dependencies for connecting to mysql to the user plugin environment.

Reply:
I think this could be shipped as an "addon" since its realistically just the installation of the mysqlclient from pip. As such, a user-plan for this would be including the mysql client from pip and the netdata plan in a single configuration plan (wrapper plan).

While its "common" functionality, its not shipped by default and we should honour what upstream is doing with their releases, so we probably shouldn't include it here. Thanks @smacfarlane for the review.

Reply:
@predominant any thoughts on how to guide users on how to make those user plans though? Would it make sense to leave the python env bits scaffolded in the plan and comment out or move to the readme the MySQL example?

My goal with this PR wasn't to give all users MySQL support, but to give all users a model for adding python deps to their version of the plan